salt/utils/decorators/state.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+'''
+Decorators for salt.state
+
+:codeauthor: :email:`Bo Maryniuk (bo@suse.de)`
+'''
+
+from __future__ import absolute_import
+from salt.exceptions import SaltException
+
+
+def state_output_check(func):
+    '''
+    Checks for specific types in the state output.
+    Raises an Exception in case particular rule is broken.
+
+    :param func:
+    :return:
+    '''
+    def _func(*args, **kwargs):
+        '''
+        Ruleset.
+        '''
+        result = func(*args, **kwargs)
+        if not isinstance(result, dict):
+            err_msg = 'Malformed state return, return must be a dict.'
+        elif not isinstance(result.get('changes'), dict):
+            err_msg = "'Changes' should be a dictionary."
+        else:
+            missing = []
+            for val in ['name', 'result', 'changes', 'comment']:
+                if val not in result:
+                    missing.append(val)
+            if missing:
+                err_msg = 'The following keys were not present in the state return: {0}.'.format(', '.join(missing))
+            else:
+                err_msg = None
+
+        if err_msg:
+            raise SaltException(err_msg)
+        return result
+
+    return _func
+
+
+def state_output_unificator(func):
```

Comment:
`unifier` makes more sense here IMO, since `unificator` isn't a word. But as long as no one else cares, I think this is fine.

Reply:
@terminalmage Right... Not English native. :laughing: Lemme fix it right now.

Reply:
👍